classes/extension/aggregator.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace mageekguy\atoum\extension;
+
+
+use mageekguy\atoum\exceptions\logic\invalidArgument;
+
+class aggregator extends \splObjectStorage
+{
+    public function getHash($object)
+    {
+        if (is_object($object) === false)
+        {
+            throw new invalidArgument(__METHOD__ . ' expects parameter 1 to be object, ' . gettype($object) . ' given');
```

Comment:
@Hywan we do not rely on the `splObjectStorage::getHash()` method here, we simply override it. We can't typehint on object so we check here.

Do you prefer a `stdClass` typehint ?